modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx

```diff
@@ -618,6 +619,16 @@ function Form (props) {
           <Typography variant="body1" paragraph>Time of the last successful save: {DateTime.fromISO(lastSaveTimestamp.toISOString()).toRelativeCalendar()}</Typography>
         }
       </ErrorDialog>
+      { isEdit && <SessionExpiryWarningModal
+                    lastSaveTimestamp={lastSaveTimestamp}
+                    saveDataWithCheckin={saveDataWithCheckin}
+                    onStay={() => saveData(undefined, false)}
+                    onExit={() => {
+                            // Redirect the user to the /
+                           // Navigating away will trigger a save with checkin
+                            props.history.push("/");
+                            }}
+                    /> }
```

Comment:
How about some indentation cleanup? Also, I don't think we need those comments.

```suggestion
      { isEdit &&
        <SessionExpiryWarningModal
          lastSaveTimestamp={lastSaveTimestamp}
          saveDataWithCheckin={saveDataWithCheckin}
          onStay={() => saveData(undefined, false)}
          onExit={() => props.history.push("/")}
        />
      }
```

Reply:
Hm, the error dialog just above should also be rendered only in edit mode...